reference/constraints/ExpressionLanguageSyntax.rst

```diff
@@ -0,0 +1,142 @@
+             * @Assert\ExpressionLanguageSyntax()
+             */
+            protected $promotion;
+
+            /**
+             * @Assert\ExpressionLanguageSyntax(
+             *     names = ['user', 'shipping_centers'],
+             *     validateNames = true
+             * )
+             */
+            protected $shippingOptions;
+        }
+
+    .. code-block:: yaml
+
+        # config/validator/validation.yaml
+        App\Entity\Order:
+            properties:
+                promotion:
+                    - ExpressionLanguageSyntax: ~
+                shippingOptions:
+                    - ExpressionLanguageSyntax:
+                        names: ['user', 'shipping_centers']
+                        validateNames: true
+
+    .. code-block:: xml
+
+        <!-- config/validator/validation.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <constraint-mapping xmlns="http://symfony.com/schema/dic/constraint-mapping"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://symfony.com/schema/dic/constraint-mapping https://symfony.com/schema/dic/constraint-mapping/constraint-mapping-1.0.xsd">
+
+            <class name="App\Entity\Order">
+                <property name="promotion">
+                    <constraint name="ExpressionLanguageSyntax"/>
+                </property>
+                <property name="shippingOptions">
+                    <constraint name="ExpressionLanguageSyntax">
+                        <option name="names">['user', 'shipping_centers']</option>
+                        <option name="validateNames">true</option>
+                    </constraint>
+                </property>
+            </class>
+        </constraint-mapping>
+
+    .. code-block:: php
+
+        // src/Entity/Student.php
+        namespace App\Entity;
+
+        use Symfony\Component\Validator\Constraints as Assert;
+        use Symfony\Component\Validator\Mapping\ClassMetadata;
+
+        class Order
+        {
+            public static function loadValidatorMetadata(ClassMetadata $metadata)
+            {
+                $metadata->addPropertyConstraint('promotion', new Assert\ExpressionLanguageSyntax());
+
+                $metadata->addPropertyConstraint('shippingOptions', new Assert\ExpressionLanguageSyntax([
+                    'names' => ['user', 'shipping_centers'],
+                    'validateNames' => true,
+                ]));
+            }
+        }
+
+Options
+-------
+
+.. include:: /reference/constraints/_groups-option.rst.inc
+
+message
+~~~~~~~
+
+**type**: ``string`` **default**: ``This value should be a valid expression.``
+
+This is the message displayed when the validation fails.
+
+names
+~~~~~
+
+**type**: ``array`` or ``null`` **default**: ``null``
+
+If the ``validateNames`` is ``true``, the expression can only use the variables
+whose names are included in this array. Use ``null`` to allow expressions to use
+any variables.
```

Comment:
While the proposal from this PR is not implemented, when `validateNames` is `true` and `names` is `null` - any variables is not allowed in validated expression.

Reply:
```suggestion
whose names are included in this array. Use ``null`` and set ``validateNames`` to ``false``
to allow expressions to use any variables.
```